perfkitbenchmarker/linux_virtual_machine.py

```diff
@@ -954,3 +956,191 @@ def MoveFile(self, target, source_path, remote_path=''):
 
     # Copies the file to its final destination in the container
     target.ContainerCopy(file_name, remote_path)
+
+
+class JujuMixin(DebianMixin):
+    """
+    Class to allow running juju-deployed workloads
+
+    Bootstraps a Juju environment using the manual provider:
+    https://jujucharms.com/docs/stable/config-manual
+    """
+
+    OS_TYPE = 'juju'
+
+    isController = False
+
+    """
+    A reference to the juju controller, useful when operations occur against
+    a unit's VM but need to be preformed from the controller.
+    """
+    controller = None
+
+    units = []
+
+    environments_yaml = """
+    default: perfkit
+
+    environments:
+        perfkit:
+            type: manual
+            bootstrap-host: {0}
+    """
+
+    def juju_bootstrap(self):
+        resp, _ = self.RemoteHostCommand("juju bootstrap")
+        pass
+
+    def juju_addmachine(self, ip):
+        resp, _ = self.RemoteHostCommand("juju add-machine ssh:%s" % ip)
+
+    def juju_configure_environment(self):
+        if self.isController:
+
+            resp, _ = self.RemoteHostCommand("mkdir -p ~/.juju")
+
+            f = tempfile.NamedTemporaryFile(delete=False)
+            f.write(self.environments_yaml.format(self.internal_ip))
+            f.close()
+            self.RemoteCopy(f.name, "~/.juju/environments.yaml")
+            os.unlink(f.name)
+        pass
+
+    def juju_environment(self, environment=''):
+        # TODO: set environment on demand
+        try:
+            output, _ = self.RemoteHostCommand('juju switch')
+            return output.strip()
+        except:
+            pass
```

Comment:
Why are you eating exceptions here (and below)? I see that you do return None if the commands fail, but it doesn't look like all of the return values get checked. I'd prefer to just let the exception get raised unless there's a good reason not to.